src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,149 @@
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order from DataBase");
+        }
+        return Optional.of(order);
+    }
+
+    @Override
+    public List<Order> getAll() {
+        String query = "SELECT * FROM orders";
+        List<Order> orders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Order order = getOrderFromResultSet(resultSet);
+                orders.add(getProductsFromOrder(order));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get orders in DataBase");
+        }
+        return orders;
+    }
+
+    @Override
+    public Order update(Order element) {
+        String query = "DELETE FROM orders_products WHERE order_id = ?";
+        String query2 = "INSERT INTO orders_products (order_id, product_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, element.getId());
+            statement.execute();
+            PreparedStatement statement2 = connection.prepareStatement(query2);
+            for (Product product : element.getProducts()) {
+                statement2.setLong(1, element.getId());
+                statement2.setLong(2, product.getId());
+                statement2.execute();
+            }
+            return element;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update order in DataBase");
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM orders WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.execute();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete order in DataBase");
+        }
+    }
+
+    private Order getOrderFromResultSet(ResultSet resultSet) throws SQLException {
+        Order order = new Order();
+        Long userId = resultSet.getLong("user_id");
+        Long orderId = resultSet.getLong("order_id");
+        order.setId(orderId);
+        order.setUser(userId);
+        return order;
+    }
+
+    private Order getProductsFromOrder(Order order) throws SQLException {
+        String query = "SELECT * FROM orders "
+                + "LEFT JOIN orders_products ON orders.order_id = orders_products.order_id "
+                + "LEFT JOIN products on products.product_id = orders_products.product_id "
+                + "WHERE orders.order_id = ?";
```

Comment:
do you need to join orders table too? or you can get it just from orders_products and products?

Reply:
same about shoppingCartDao

Reply:
Thanks